pkg/square/square.go

```diff
@@ -48,8 +48,9 @@ func Build(txs [][]byte, appVersion uint64, maxSquareSize int) (Square, [][]byte
 //
 // Note that this function does not check the underlying validity of
 // the transactions.
-func Construct(txs [][]byte, appVersion uint64, maxSquareSize int) (Square, error) {
-	builder, err := NewBuilder(maxSquareSize, appconsts.SubtreeRootThreshold(appVersion), txs...)
+func Construct(txs [][]byte, appVersion uint64) (Square, error) {
+	maxSquareSize := appconsts.SquareSizeUpperBound(appVersion)
+	builder, err := NewBuilder(maxSquareSize, appVersion, txs...)
 	if err != nil {
```

Comment:
we no longer have a reason to pass the square size to the construct method, which should always use the value based on the appversion

Reply:
this was incorrect!!! #2015